vector.go

```diff
@@ -124,6 +124,13 @@ func (v VectorType) Unmarshal(data []byte, value interface{}) error {
 	}
 	rv = rv.Elem()
 	t := rv.Type()
+	if t.Kind() == reflect.Interface {
+		if t.NumMethod() != 0 {
+			return unmarshalErrorf("can not unmarshal into non-empty interface %T", value)
+		}
+		t = reflect.TypeOf(v.Zero())
```

Comment:
This is assuming that the interface holds that type but it might not, I think you need to instead check if it's nil and then if its not nil, then do `rv = rv.Elem() t = rv.Type()`